cinder/client.go

```diff
@@ -0,0 +1,264 @@
+package cinder
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type TokenFn func() string
+
+func SetEndpointFn(endpoint *url.URL, wrappedHandler RequestHandlerFn) RequestHandlerFn {
```

Comment:
Seeing as the Host field is the only part of `endpoint` that is used, just pass in host?